### DISCUSSION TITLE
当加入稍后读后，自动发送到你的 Telegram Channel

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3092] [服务]
> 手动导出到 Telegram Channel

https://user-images.githubusercontent.com/81074/142806669-dca1553f-9d32-4eba-85f1-5592e82d9b93.mp4

> 当加入稍后读后，自动发送到你的 Telegram Channel

https://user-images.githubusercontent.com/81074/142812726-3a126361-7aea-4521-90d1-9141066ec2a1.mp4

# 前置知识

> 需要事先完成下面的配置。

1. [如何开通开放平台](https://github.com/Kenshin/simpread/discussions/2757)
2. [API 快照功能](https://github.com/Kenshin/simpread/discussions/2994)
3. [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)

# integromat

> 如果你用过 IFTTT / Zapier 的话，integromat 是一个比 IFTTT 更强大的自动化程序，更难能可贵的是，免费版的即可使用 Webhook 服务。（ _Zapier 的 Webhook 收费，虽然 IFTTT 不收费，但它的定制化无法跟 integromat 相比_） 

> 此教程需要你事先注册 https://www.integromat.com/

# 配置

> 使用 [@BotFather](t.me/BotFather) 新建一个 bot

![iShot2021-11-19 16 22 39](https://user-images.githubusercontent.com/81074/142805294-c3967239-60ce-406d-bec5-71a6db3a5033.png)

> 建立一个公开的 Telegram Channel  并把上步的 bot 添加进来赋予管理员权限

![image](https://user-images.githubusercontent.com/81074/142802268-a7d0561a-35d7-4723-91c8-a57e71266ad3.png)

## 配置 integromat

> 新建一个 Webhook

![image](https://user-images.githubusercontent.com/81074/142802935-d98c237e-b719-43da-80b5-edfb5266cf16.png)

> 新建一个 Telegram

![image](https://user-images.githubusercontent.com/81074/142803513-4eac2002-0db2-445f-91b9-451e809ec526.png)

> 详细设置

![image](https://user-images.githubusercontent.com/81074/142803502-3a06802a-4db6-46b0-ab09-402ee91f5798.png)

```
来自 [简悦](http://simpread.pro/) 的收藏

▎ {{1.desc}}

[{{1.title}}]({{1.url}})

{{if(1.tags = ""; ; "#" + 1.tags)}}
```

> 其它设置，可以根据下图的设置，包括：

- 当 channel 得到信息后是否提示
- 是否显示 Link Previews

![image](https://user-images.githubusercontent.com/81074/142803733-f28f1fb6-2d44-4c97-8477-66727c9e5aff.png)

> 设置完毕

![image](https://user-images.githubusercontent.com/81074/142803905-97bb3d7b-008f-4cc3-8d02-316b6c67d27e.png)

## 简悦扩展端设置 Webhook

> 将上步得到的 integromat webhook 替换为下面的地址

```
{
    "name": "Webhook2Telegram",
    "url": "https://hook.integromat.com/******",
    "type": "POST",
    "body": {
        "url": "{{url}}",
        "title": "{{title}}",
        "desc": "{{desc}}",
        "tags": "{{tags}}",
        "content": "{{content}}"
    }
}
```

> 将上面的代码使用 https://tool.lu/json/ 提供的压缩方式压缩一行，粘贴到 选项页 → 服务 → Webhook

![image](https://user-images.githubusercontent.com/81074/142805662-f7345a9d-1fc9-48c8-b20e-c1c69a0b574d.png)

> 至此，完全全部配置。

# 测试

> 回到 integromat 并开始测试

![image](https://user-images.githubusercontent.com/81074/142806154-b3ee4485-c329-48f7-8e35-41864d3c92ac.png)

> 打开 https://sspai.com/post/69972 并按照下面的视频操作，如果跟视频的效果类似，则说明适配成功

https://user-images.githubusercontent.com/81074/142806221-e792bea0-3403-4b89-bc30-ba2012bf44f4.mp4

> 在你的 Telegram Channel 就会收到下图的内容

![image](https://user-images.githubusercontent.com/81074/142806371-d7548df9-0313-491f-9d61-bfb5a066360b.png)

> 最后，开启此 Scenario 完成全部配置

![image](https://user-images.githubusercontent.com/81074/142806481-9a8ecfda-b99f-474c-85b9-68c16c759180.png)

# 更多玩法

> 第一个视频是通过手动触发导出动作完成的，你有也可以将这个服务加入到稍后读，当加入稍后读后自动触发

![image](https://user-images.githubusercontent.com/81074/142811089-bf2b081c-b5b8-4c34-8c22-218982d87e0d.png)

# 与 Telegram 相关的其它用法

- [当加入稍后读后，自动发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3092)

- [保存到 Telegraph](https://github.com/Kenshin/simpread/discussions/2795)

- [适配了一组关于 Telegram 的新站](https://github.com/Kenshin/simpread/discussions/2883)

- [将当前阅读模式转换为 Telegraph Page 并发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3337)

# 关联

- [利用 Airtable 实现属于自己的链接公开分享集 #3003](https://github.com/Kenshin/simpread/discussions/3003)
- [Inoreader 加星后自动保存到简悦的稍后读以及生成本地快照 #3007](https://github.com/Kenshin/simpread/discussions/3007)
- [Raindrop.io 加星后自动导入到简悦并生成本地快照 #3021](https://github.com/Kenshin/simpread/discussions/3021)
- [加入 Pocket 后自动保存到简悦的稍后读（包含快照） #3024](https://github.com/Kenshin/simpread/discussions/3024)
- [Instapaper 加星后自动保存到简悦的稍后读（包含快照） #3034](https://github.com/Kenshin/simpread/discussions/3034)
- [将简悦变成你的 RSS 阅读器（自动下载快照到本地） #3036](https://github.com/Kenshin/simpread/discussions/3036)
- [保存到 Pinboard.in 同时也保存到简悦（包括本地快照） #3048](https://github.com/Kenshin/simpread/discussions/3048)
- [订阅 Newsletter 后自动保存到简悦的稍后读 #3088](https://github.com/Kenshin/simpread/discussions/3088)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）#3104](https://github.com/Kenshin/simpread/discussions/3104)
- [保存 RSS 的全文到简悦的稍后读（包括快照） #3186](https://github.com/Kenshin/simpread/discussions/3186)

# 引申

- [[教程\]如何开通开放平台 #2757](https://github.com/Kenshin/simpread/discussions/2757)
- [当在开放平台设置完毕后，如何测试配置是否成功？ #3020](https://github.com/Kenshin/simpread/discussions/3020)
- [授权登录（如何使用第三方登录简悦的高级账户） #1570](https://github.com/Kenshin/simpread/discussions/1570)
- [个人中心 #2959](https://github.com/Kenshin/simpread/discussions/2959)
- [免密码登录 #2974](https://github.com/Kenshin/simpread/discussions/2974)

**Comment by xooooooooox** [comment]
有没有telegram的学习贴啊，现在除了会使用加入电报群功能，其它一概不知（打扰了）

**Reply by Kenshin** [reply]
下周有篇文章，应该适合你。
